client/transaction-pool/graph/src/validated_pool.rs

```diff
@@ -159,7 +158,7 @@ impl<B: ChainApi> ValidatedPool<B> {
 	}
 
 	/// Submit single pre-validated transaction to the pool.
-	fn submit_one(&self, tx: ValidatedTransactionFor<B>) -> Result<ExHash<B>, B::Error> {
+	fn submit_one(&self, tx: ValidatedTransactionFor<B>) -> Result<BlockHash<B>, B::Error> {
```

Comment:
Why would you change something that denotes transaction hash to type alias named `BlockHash`?

Reply:
Explained above. Nothing changed here.

Reply:
> > Cannot we prepare all of those retracted blocks in advance and just add them to the current event to the retracted list?
> 
> Where should that be done?

The place where `ChainEvent::NewBlock` is instantiated:
https://github.com/paritytech/substrate/pull/6123/files#diff-240c17a1e4dbce4ad137b7777bf03ce4R1050

> 
> > We request them from transaction pool every time anyway, don't we?
> 
> What do we request from the transaction pool?

List of blocks whose transaction are going to be resubmitted